### DISCUSSION TITLE
How do I use the between operator?

### DISCUSSION THREAD

**Discussion by mikkel-arturo** [sensedeep/dynamodb-onetable#374] [Q&A] [answered]
I am attempting a query like so (communityId is the hash key, publish timestamp is the sort key - gsi1SK)
```
Post.find(
			{
				communityId,
				publishTimestamp: {
					between: [startTime, endTime],
				},
			},
			{
				index: 'gsi1',
				next: options.next,
			},
		)
```
And it is throwing the error `OneTableError: Value for "gsi1SK" is not a primitive value`. I see in the code that it is testing `if (typeof v == 'object' && v.toString() == '[object Object]')` and since v is `between: [startTime, endTime]` then it is triggering. Am I doing something wrong? This has intellisense support and everything so it seems to be correct.

**Comment by mobsense** [comment, marked as answer]
That is strange as that code should only be invoked when there is a ${var} in the value to expand value templates.

Can you create a test case that demonstrates and raise a bug?
